### DISCUSSION TITLE
Multiconection

### DISCUSSION THREAD

**Discussion by m2teusz** [syssi/esphome-tianpower-bms#36] [General]
Hi, is it possible to connect two TP-LT55AT devices to one ESP via BT?

**Comment by syssi** [comment]
Yes. It's possible but in worst case the performance/stability of the setup degrades a bit.

**Reply by m2teusz** [reply]
Do you have an example yaml file?

**Reply by syssi** [reply]
Are you familiar with ESPHome or is this your first ESPHome based project?

This is another project using the `ble_client` component multiple times. You have to replicate this pattern: https://github.com/syssi/esphome-jk-bms/blob/main/esp32-heltec-balancer-ble-example-multiple-devices.yaml

If you aren't successful I will provide a dedicated example this evening.

**Reply by m2teusz** [reply]
Thank you for the example, that's enough for me. As I thought, you should make a clone of the code for the next MAC address. I already have a working ESP for one piece of BMS.
![bms-ble](https://github.com/user-attachments/assets/32eb2c11-9f8c-4e92-86b3-fb59c3051133)

**Reply by syssi** [reply]
I've pushed an example how to access two devices using the same ESP32 to the project: https://github.com/syssi/esphome-tianpower-bms/pull/37/files
